tools/azsdk-cli/docs/specs/0-scenario-2.spec.md

```diff
@@ -0,0 +1,957 @@
+   - **Open Question**: Should there be a dedicated packing/packaging tool?
+     - Packing step is currently missing from the tools
+     - Not needed for local development scenarios
+     - Required for pipeline scenarios to produce distributable artifacts
+     - Decision needed: Add dedicated `azsdk_package_pack` tool or integrate packing into existing tools
+
+---
+
+## Implementation Plan
+
+Focus this scenario’s delivery on producing high-quality MCP/CLI tool specifications and deterministic flows:
+
+1. Author detailed tool specs for new Scenario 2 tools:
+   - Enhanced `azsdk_verify_setup` (with optional fix mode for remediation)
+   - Enhanced `azsdk_customized_code_update` (unified two-phase customization workflow)
+   - Enhanced `azsdk_package_run_tests` (unified testing with automatic resource management)
+2. Confirm integration points with existing tools (`azsdk_package_generate_code`, `azsdk_package_build_code`).
+3. Define telemetry events (start, success, failure) and common correlation identifiers across all tools.
+4. Document unified customization workflow (Phase A TypeSpec → regenerate → Phase B code → build verification).
+5. Document testing flows (automatic resource provisioning, test mode selection, recording capture) for both net-new and existing SDKs.
+6. Finalize prompt catalog revisions and CLI command adjustments (consolidated commands, simplified options).
+7. Update shared docs (`mcp-tools.md`, engineering hub) to include updated tools and flows.
+8. Identify any edge cases for environment remediation (unsupported auto-install) and test asset generation complexity to feed into Open Questions.
+
+---
+
+## Documentation Updates
+
+- Update the MCP tools list in `mcp-tools.md` with all new tools introduced in Scenario 2
+- Update Eng Hub documentation with Scenario 2 workflow, customization guidance, and testing patterns
+
+---
+
+## Metrics/Telemetry
+
+- Collect telemetry for all MCP tools and CLI commands introduced in Scenario 2
+- Track tool usage patterns, success/failure rates, and execution duration to identify opportunities for improvement
+
+---
+
+## Open Questions
+
+### Scope Considerations
+
+**Net-New SDKs:**
+
+- Should [Net-New SDK](#net-new-sdk) support be included in Scenario 2, or deferred to a later scenario?
+- Current assessment: Generation and testing stages would require additional work for Net-New SDKs
+  - Generation: Project scaffolding and bootstrapping for languages
+  - Testing: Test asset creation is complex and may require manual configuration due to service-specific resource requirements
+- Decision needed: Include Net-New SDK support now, or focus on existing SDK workflows and add Net-New support in a future scenario?
+
+### Tool and Process Questions
+
+**Tool Auto-Installation Rules:**
+
+- What are the specific criteria for determining which tools can be auto-installed/auto-upgraded by `azsdk_verify_setup` versus which require manual installation with guidance links?
```

Comment:
My two cents, anything touching the system should be consider out of scope, as the agent shouldn't have admin permissions. Installing Python? Too hard. Install tsp-client for the current user? Good